packages/core/src/views/notify/Notification.svelte

```diff
@@ -2,32 +2,149 @@
   import StatusIconBadge from './StatusIconBadge.svelte'
   import NotificationContent from './NotificationContent.svelte'
   import { removeNotification } from '../../store/actions'
-
   import type { Notification } from '../../types'
   import closeIcon from '../../icons/close-circle'
-
-  import { chainStyles, networkToChainId } from '../../utils'
   import { onDestroy, onMount } from 'svelte'
   import { configuration } from '../../configuration'
+  import { removeTransaction, transactions$, wallets$ } from '../../streams'
+  import type { Network } from 'bnc-sdk'
+  import { BigNumber } from 'ethers'
+  import { state } from '../../store'
 
-  const { device } = configuration
+  import {
+    chainStyles,
+    gweiToWeiHex,
+    networkToChainId,
+    toHexString
+  } from '../../utils'
+
+  const { device, gas, apiKey } = configuration
 
   export let notification: Notification
   export let updateParentOnRemove: () => void
 
   let timeoutId: NodeJS.Timeout
+  let hovered = false
+
+  // get transaction based on notification id
+  const transaction = transactions$
+    .getValue()
+    .find(({ hash }) => hash === notification.id)
+
+  // get wallet that sent transaction
+  const wallet =
+    transaction &&
+    $wallets$.find(
+      ({ accounts }) =>
+        !!accounts.find(
+          ({ address }) =>
+            address.toLowerCase() === transaction.from.toLowerCase()
+        )
+    )
 
   onMount(() => {
     if (notification.autoDismiss) {
       timeoutId = setTimeout(() => {
         removeNotification(notification.id)
+        removeTransaction(notification.id)
       }, notification.autoDismiss)
     }
   })
 
   onDestroy(() => {
     clearTimeout(timeoutId)
   })
+
+  function actionableEventCode(eventCode: Notification['eventCode']): boolean {
+    switch (eventCode) {
+      case 'txPool':
+        return true
+      default:
+        return false
+    }
+  }
+
+  function validGasNetwork(network: Network) {
+    switch (network) {
+      case 'main':
+      case 'matic-main':
+        return true
+      default:
+        return false
+    }
+  }
+
+  function walletSupportsReplacement() {
+    if (!wallet) return false
+
+    switch (wallet.label) {
+      case 'Ledger':
+      case 'Trezor':
+      case 'Keystone':
+      case 'KeepKey':
+        return true
+      default:
+        return false
+    }
+  }
```

Comment:
@aaronbarnardsound maybe a little cleaner/simpler? 

```suggestion
const WALLETS_SUPPORT_REPLACEMENT = ['Ledger', 'Trezor', 'Keystone', 'KeepKey']
const walletSupportsReplacement = () => wallet && wallet.label in WALLETS_SUPPORT_REPLACEMENT
```

Reply:
I have simplified based on this idea and have removed my cherished `switch` statements. 

By the way the `in` operator does not work when checking whether a `string` value is in an `array`. It is more of an object parameter lookup.